symforce/slam/imu_preintegration/imu_factor.h

```diff
@@ -0,0 +1,82 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <Eigen/Dense>
```

Comment:
I never quite understand why we include `<Eigen/Dense>`. It pulls in some other headers like QR, SVD that are not needed.

Reply:
I did check and `<Eigen/Dense>` does always get included (from `<sym/rot3.h>`), but I figured I might as well start getting in the habit of using more minimal headers, so I've made the change for the new files added by this PR (obviously, this won't really matter unless we make the changes elsewhere as well though).

Reply:
@chao-qu-skydio have you observed compile time differences from including the more minimal headers instead of `Eigen/Dense`?

Reply:
https://artificial-mind.net/projects/compile-health/
![image](https://user-images.githubusercontent.com/113060634/210449550-fa47796f-5b6b-4000-9afe-9814d8bb068f.png)

Here it shows that the difference is slightly less than 100ms. However, the total added compile time also depends on how many translation units we need to build. 

I think including `<Eigen/Dense>` is fine if we are already doing it, but it's just something I don't do personally.

Reply:
Cool, yeah we could probably change most of those to Eigen/Core, I think that's a good idea

Reply:
And support doing that for the change in this PR